google/cloud/BUILD.bazel

```
@@ -75,6 +75,17 @@ cc_library(
             "GOOGLE_CLOUD_CPP_HAVE_OPENTELEMETRY",
         ],
         "//conditions:default": [],
+    }) + select({
+        "@platforms//os:windows": [
+            "WIN32_LEAN_AND_MEAN",
+        ],
+        "//conditions:default": [],
+    }),
+    linkopts = select({
```

As a possible workaround for 

https://github.com/bazelbuild/bazel/issues/15024

```suggestion
    local_defines = select({
      "@platforms//os:windows": [
        "_WIN32_WINNT=0x0A00",
      ],
      "//conditions:default": [],
    }),
    linkopts = select({
```

We should also consider using `local_defines` for `WIN32_LEAN_AND_MEAN`.